File: Blog.Data/Models/Comment.cs
using Blog.Data.Identity.Models;
using System;
using System.ComponentModel.DataAnnotations;

namespace Blog.Data.Models
{
    public class Comment
    {
        public int CommentId { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public int RecordId { get; set; }

        public Record Record { get; set; }
    }
}

File: Blog.ReactUI/Src/Types/Comments/Comment.ts
import Record from "Types/Records/Record";
import User from "Types/Account/User";

export default class Comment {
  public CommentId: number;

  public Content: string;

  public CreateDate: Date;

  public RecordId: number;

  public Record: Record;

  public UserId: number;

  public User: User;
}
